### ISSUE TITLE
Error: A web3 provider is not attached to a window.

### ISSUE THREAD

**Issue by abcoathup** [OpenZeppelin/openzeppelin-network.js#16]
When running a simple React app with OpenZeppelin Network.js in a browser without injected Web3 (e.g. MetaMask extension off) the following error is displayed: `Error: A web3 provider is not attached to a window.`

Is this an issue with OpenZeppelin Network.js or is there a pattern that should be used to detect injected web3 not present?

## Simple React App
```js
import React from 'react';
import './App.css';

import { useWeb3Injected } from '@openzeppelin/network/react';

function App() {

  const web3Context = useWeb3Injected();

  return (
    <div className="App">

      <div>
        <div>
          Hello Web3
        </div>
      </div>
    </div>
  );
}

export default App;
```

## Error
<img width="704" alt="No web3 provider" src="https://user-images.githubusercontent.com/28278242/66101610-d894fd80-e5f2-11e9-828f-6b18e942129c.png">

**Comment by abcoathup** [comment]
Originally created as an Issue on GSN Starter Kit: https://github.com/OpenZeppelin/starter-kit-gsn/issues/14

Moving to here as specific to OpenZeppelin Network.js

**Comment by ylv-io** [comment]
Hey @abcoathup. It is an expected behavior. If user doesn't have an injected provider like MetaMask, an error will be thrown and user expected to handle it.

**Comment by abcoathup** [comment]
Should be fixed by #24
